lua/telescope/builtin/internal.lua

```diff
@@ -231,9 +231,35 @@ internal.loclist = function(opts)
 end
 
 internal.oldfiles = function(opts)
-  local results = vim.tbl_filter(function(val)
-    return 0 ~= vim.fn.filereadable(val)
-  end, vim.v.oldfiles)
+  opts.include_current_session = utils.get_default(opts.include_current_session, true)
+
+  local results = {}
+
+  if opts.include_current_session then
+    for _, buffer in ipairs(vim.split(vim.fn.execute(':buffers! t'), "\n")) do
```

Comment:
Because if you close a buffer, it'll get removed from the list, which doesn't make sense. It's the same as my `remove_buf()` argument here https://github.com/nvim-telescope/telescope.nvim/issues/433#issuecomment-798831113.

...It does a file existance check though, so it never lists scratch buffers, etc. It actually lists the exact same files as `v:oldfiles` does after you restart Vim. I made sure of that!

Reply:
@kristijanhusak Because if you manually open a file buffer, then `:bdelete` to close that buffer, it'll get removed from the finder until you restart Vim and see it in `v:oldfiles`, which doesn't make any sense. It's the same as my `remove_buf()` argument from our other thread here https://github.com/nvim-telescope/telescope.nvim/issues/433#issuecomment-798831113...

>Whether or not I close a buffer, I would still expect it to show in `:History` or `:Telescope oldfiles`, since I did in fact open it. If I restart Vim, it would be in the oldfiles list, so why not [show that closed/unlisted buffer] in the list?

That said, it does a file existence filter check though, so it never lists temporary scratch buffers, terminal buffers, etc. It actually lists the exact same results as `v:oldfiles`, just that you don't have to restart Vim to see the old files in your current session. I made sure of that!

Reply:
Ok, makes sense, thanks!